client/blocks/login/login-form.jsx

```diff
@@ -97,25 +110,90 @@ export class LoginForm extends Component {
 				this.setState( { usernameOrEmail: nextProps.socialAccountLinkEmail } );
 			}
 		}
+
+		if ( this.props.hasAccountTypeLoaded && ! nextProps.hasAccountTypeLoaded ) {
+			defer( () => this.usernameOrEmail && this.usernameOrEmail.focus() );
+		}
+
+		if ( ! this.props.hasAccountTypeLoaded && isRegularAccount( nextProps.accountType ) ) {
+			defer( () => this.password && this.password.focus() );
+		}
+
+		if ( ! this.props.hasAccountTypeLoaded && isPasswordlessAccount( nextProps.accountType ) ) {
+			this.props.recordTracksEvent( 'calypso_login_block_login_form_send_magic_link' );
+
+			this.props.fetchMagicLoginRequestEmail( this.state.usernameOrEmail )
+				.then( () => {
+					this.props.recordTracksEvent( 'calypso_login_block_login_form_send_magic_link_success' );
+				} )
+				.catch( error => {
+					this.props.recordTracksEvent( 'calypso_login_block_login_form_send_magic_link_failure', {
+						error_code: error.code,
+						error_message: error.message,
+					} );
+				} );
+
+			page( login( { isNative: true, twoFactorAuthType: 'link' } ) );
+		}
 	}
 
 	onChangeField = event => {
 		this.props.formUpdate();
+
 		this.setState( {
 			[ event.target.name ]: event.target.value,
 		} );
 	};
 
-	onSubmitForm = event => {
+	getAuthAccountType() {
+		const { usernameOrEmail } = this.state;
+
+		this.props.recordTracksEvent( 'calypso_login_block_login_form_get_auth_type' );
+
+		this.props
+			.getAuthAccountType( usernameOrEmail )
+			.then( () => {
+				this.props.recordTracksEvent( 'calypso_login_block_login_form_get_auth_type_success' );
```

Comment:
Seems like tracking have to be off-loaded to the data layer, since we can't get promise from here.
[example](https://github.com/Automattic/wp-calypso/blob/master/client/state/data-layer/wpcom/sites/automated-transfer/initiate/index.js#L74)

Reply:
Thanks for the pointer! I'll update the code accordingly.